src/data_store/jag_io.cpp

```diff
@@ -0,0 +1,386 @@
+#include "lbann_config.hpp" // may define LBANN_HAS_CONDUIT
+
+#ifdef LBANN_HAS_CONDUIT
+
+#include "lbann/data_store/jag_io.hpp"
+#include "lbann/data_readers/data_reader_jag_conduit.hpp"
+#include "lbann/utils/exception.hpp"
+#include "lbann/utils/file_utils.hpp"
+#include "lbann_config.hpp" 
+#include <unordered_map>
+#include <unordered_set>
+#include <fstream>
+#include <cstdlib>
+#include <algorithm>
+
+namespace lbann {
+
+jag_io::~jag_io() {
+  for (size_t i=0; i<m_data_streams.size(); i++) {
+    if (m_data_streams[i].is_open()) {
+      m_data_streams[i].close();
+    }  
+  }
+}
+
+jag_io::jag_io() {}
+
+void jag_io::get_hierarchy(conduit::Node &nd, std::string parent) {
+  std::string parent_2 = parent;
+  if (parent.find('/') != std::string::npos) {
+    // hack to discard keys that vary between samples;
+    // will fix later, when we have the samples we're going
+    // to actually use, and guidance as to which outputs
+    // we should use.
+    if (parent.find("outputs/scalars") == std::string::npos) {
+      m_keys.push_back( parent.substr(2));
+    }  
+  } 
+  conduit::Node nd2 = nd[parent.c_str()];
```

Comment:
c-string will automatically converted to std::string. So, there is no problem.
However, the interfaces expects "const std::string&" as below.
  Node             &operator[](const std::string &path);
  const Node       &operator[](const std::string &path) const;